python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1467,17 +1473,17 @@ def get_attributes_dict(self) -> Dict[str, Any]:
             selected_asset_keys=self._selected_asset_keys,
             can_subset=self._can_subset,
             resource_defs=self._resource_defs,
-            group_names_by_key=self._group_names_by_key,
-            metadata_by_key=self._metadata_by_key,
-            freshness_policies_by_key=self._freshness_policies_by_key,
-            auto_materialize_policies_by_key=self._auto_materialize_policies_by_key,
+            group_names_by_key=group_names_by_key,
+            metadata_by_key=metadata_by_key,
+            freshness_policies_by_key=freshness_policies_by_key,
+            auto_materialize_policies_by_key=auto_materialize_policies_by_key,
             backfill_policy=self._backfill_policy,
-            descriptions_by_key=self._descriptions_by_key,
+            descriptions_by_key=descriptions_by_key,
             check_specs_by_output_name=self._check_specs_by_output_name,
             selected_asset_check_keys=self._selected_asset_check_keys,
-            owners_by_key=self._owners_by_key,
-            tags_by_key=self._tags_by_key,
-            is_subset=self._is_subset,
+            owners_by_key=owners_by_key,
+            tags_by_key=tags_by_key,
+            is_subset=self.is_subset,
```

Comment:
Aren't a lot of these `by_key` dicts already properties you could just read?

Also, perhaps best for a follow-up but it's unfortunate that we have to round-trip through this `*_by_key` pattern when copying. Would it be possible to pass specs directly to the constructor instead (as an alternative to the by_key form).

Reply:
> Aren't a lot of these by_key dicts already properties you could just read?

Yes, but each of those properties requires iterating over the `specs_by_key` dict. So using them would require iterating over that dict once per property, instead of just once.

And I'd like to eventually eliminate those properties (which are private) and expect people to go through the specs.